File: guests/ocaml/split.ml
(* The split scene, OCaml port — guests/rust/split.rs,
   tools/scenes/split.steps. *)

open Kaya_app

let detail = 7L

let () =
  let app = Kaya_app.create () in

  build app (fun () ->
     window ~title:"split" ~panes:2 ();
     let s = signal_str ("list pane") in
     let on_detail () =
       push_entry ~title:"detail"
         ~on_popped:(fun () -> write s ("popped detail"))
         detail;
       (let caption = signal_str ("detail pane") in
        let pane = column [ label ~a11y_id:"detail" ~bind:caption ] () in
        mount_in detail pane)
     in
     let root =
       column
         [
           (* Authored ids: an index read passes whether or not anything
              reached the screen. *)
           label ~a11y_id:"list" ~bind:s (* label#0 *);
           button ~text:"open detail" ~on_click:on_detail;
         ]
         ()
     in
     mount root);

  exit (run app)

File: guests/swift/split.swift
// The split scene, Swift port — guests/rust/split.rs,
// tools/scenes/split.steps.

import Foundation
import Kaya

let DETAIL: UInt64 = 7

let app = KayaApp()

app.build { tx in
    tx.window(title: "split", panes: 2)
    let status = tx.signal(.str("list pane"))
    let root = tx.column { root in
        // Authored ids: an index read passes for an arm that drew nothing.
        tx.setA11yId(tx.label(bind: status), "list")  // label#0
        tx.button(
            "open detail",
            onClick: { inner in  // button#0
            inner.pushEntry(
                DETAIL, title: "detail",
                onPopped: { tx2 in tx2.write(status, .str("popped detail")) })
            let pane = inner.column { pane in
                let caption = inner.signal(.str("detail pane"))
                inner.setA11yId(inner.label(bind: caption), "detail")
                return pane
            }
            inner.mountIn(DETAIL, pane)
            })
        return root
    }
    tx.mount(root)
}

app.run()
